plugins/dota.py

```diff
@@ -0,0 +1,175 @@
+                        player_name=user,
+                        level=p['level'],
+                        kills=p['kills'],
+                        deaths=p['deaths'],
+                        assists=p['assists'],
+                        gpm=p['gold_per_min'],
+                        xpm=p['xp_per_min'],
+                        team='R' if p['player_slot'] <= 5 else 'D',
+                    )
+                    # Don't print out duplicates if we have multiple nicknames
+                    # associated with the same Steam ID
+                    break
+        return ret
+
+    def get_match_str(self, match_id):
+        if not isinstance(match_id, int):
+            return "Game not found."
+        m = self.api.get_match_details(match_id)
+        return "Dota 2 {match_id} {time}, score: {radiant_score}-{dire_score}, {winner} victory, duration {duration}m. {our_scores} {dotabuff}".format(
+            time=datetime.datetime.fromtimestamp(m['start_time']).strftime('%b %d'),
+            match_id=match_id,
+            duration=m['duration']//60,
+            dotabuff="http://www.dotabuff.com/matches/{}".format(match_id),
+            radiant_score=m['radiant_score'],
+            dire_score=m['dire_score'],
+            winner="Radiant" if m['radiant_win'] else "Dire",
+            our_scores=self.get_our_scores(m),
+        )
+
+    def _get_latest_match_id(self, steamid):
+        try:
+            matches = self.api.get_match_history(steamid)
+        except dota2api.exceptions.APIError:
+            raise "Failed to fetch from API"
+        except dota2api.exceptions.APITimeoutError:
+            raise "Connection to Dota 2 API timed out."
+        latest_match = {'match_id': 'Not found.'}
+        if 'total_results' in matches and matches['total_results'] > 0:
+            latest_match = matches['matches'][0]
+        self._update_latest_match_id(steamid, latest_match['match_id'])
+        return latest_match['match_id']
+
+    @botologist.plugin.command('dota')
+    def dota(self, cmd):
+        '''
+        Gets the latest dota match for the user requesting it.
+        '''
+        if not self._nick_exists(cmd.user.nick):
+            return "Steam ID not found for {user}.".format(
+                user=str(cmd.user.nick)
+            )
+        steamid = self._get_steam_id(cmd.user.nick)
+        if not steamid:
+            return "Invalid Steam ID."
+        latest_match_id = self._get_latest_match_id(steamid)
+        return self.get_match_str(latest_match_id)
+
+
+    @botologist.plugin.command('steamid')
+    def steamid(self, cmd):
+        '''
+        Adds the user's steam id (dota version) to the sqlite db
+        '''
+        if len(cmd.args) != 1:
+            return "Usage: !steamid 13371337"
+        try:
+            steamid = int(cmd.args[0])
+        except ValueError:
+            return "That's not a valid Steam ID."
+        if not self._add_steamid(cmd.user.nick, steamid):
```

Comment:
Maybe use `cmd.user.identifier` instead, to account for people randomly changing nicks?